WordPress/WordPressTest/ActivityContentFactoryTests.swift

```diff
@@ -15,6 +15,6 @@ final class ActivityContentFactoryTests: XCTestCase {
     }
 
     private func getDictionaryFromFile(named fileName: String) -> [String: AnyObject] {
-        return contextManager.object(withContentOfFile: fileName) as! [String: AnyObject]
+        return JSONLoader().loadFile(named: fileName) ?? [:]
```

Comment:
Initially I made the new `JSONLoader` function and the new `NSManagedObjectContext` fixture method to throw, this change of course required all the call sites to keep throwing the error all the way up to test methods, which would help us diagnosing test failures. But I got lazy 😅, 'cause there are so many places need to change.

I think I'm going to do this properly this time - propagating errors from JSON parsing function all the way up to test methods, but also updating bunch of loading test data functions along the way, like [this one here](https://github.com/wordpress-mobile/WordPress-iOS/blob/112b352938ecbea191addd94be7f61d8cd4d5bf4/WordPress/WordPressTest/ActivityLogTestData.swift).

Reply:
To optimise the error handling here and also [the loading JSON file function](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478#discussion_r862585396), I introduced [a new way](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478/files#diff-f3425ef339c7340197f30a8859ef98c5abfd92d8d13911940a4d426378c908b0) to create test objects from JSON files: https://github.com/wordpress-mobile/WordPress-iOS/pull/18478/commits/e9c69065efa4763e7681c04704d61d5b6b2847c8. I've update [notifications](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478/files#diff-4f9f4eadc7c4b2cb2a8677b7e9f9459fc70111e6e39228adb0cf40d3c299120c) related unit test to use this new pattern, to showcase how it's used in practice. If you think this is a good direction to go, I can delete the JSONLoader, and apply this new pattern to all unit tests, maybe in a separate PR, since there are going to be lots of changes.

Reply:
> I introduced [a new way](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478/files#diff-f3425ef339c7340197f30a8859ef98c5abfd92d8d13911940a4d426378c908b0) to create test objects from JSON files 

Thank you for taking the time. I believe there's merit in the approach you're proposing, I particularly like the idea of a hierarchical name-spaced fixtures grab bag to help devs get the test data they need thanks to Xcode's autocompletion. **But I'm afraid I'll have to push back on this**.

Changing how we do something as foundational and widespread as loading fixture data is something we should approach with a lot of thoughtfulness. I'd like to see an RFC post for it, a code spike in line with what you did in https://github.com/wordpress-mobile/WordPress-iOS/commit/e9c69065efa4763e7681c04704d61d5b6b2847c8, and then an announcement post to let the product devs know about it.

Going back on your current proposal leaves the question of how to move forward to avoid tests failing in non-straightforward way 🤔  I left a [suggestion](https://github.com/wordpress-mobile/WordPress-iOS/pull/18478#discussion_r863479696) in the thread about returning an `Optional` vs `throwing`.

Reply:
Good call, starting with a RFC makes sense to me 👍 . I'll remove the fixture change out of this PR.